### PR TITLE
MULE-18419: Race condition in FluxSink when events are buffered and the

### PR DESCRIPTION
**Pull request by fsgonz**
flux is completed.

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java [hunk 1]
```diff
@@ -33,23 +33,25 @@
   @Override
   public void accept(FluxSink<T> fluxSink) {
     FluxSinkRecorderDelegate<T> previousDelegate = this.delegate;
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java:35]: shouldn't this be synchronized as well?

File: core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java [hunk 2]
```diff
@@ -33,23 +33,25 @@
...
-    delegate = new DirectDelegate<>(fluxSink);
-    previousDelegate.accept(fluxSink);
+    synchronized (this) {
+      delegate = new DirectDelegate<>(fluxSink);
+      previousDelegate.accept(fluxSink);
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java:38]: shouldn't this be outside the critical region?

File: core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java [hunk 3]
```diff
@@ -33,23 +33,25 @@
...
+    }
   }
 
-  public FluxSink<T> getFluxSink() {
+  public synchronized FluxSink<T> getFluxSink() {
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java:42]: have you meassured the perf impact of synchonizing all these methods?

Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java:42]: I'm on it. I've run the tests in 4.2.2

File: core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java (hunk 2 above)
Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java:38]: This is the change that prevents the race conditon.

File: core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java (hunk 1 above)
Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/rx/FluxSinkRecorder.java:35]: It is synchronized inside the method.

**Pull request closed without merging**